File: src/TestDecls.java
public class TestDecls
{
    private static final int MINUSFIVE = -5;
    private static final int TEN = 10;

    private static enum greek {ALPHA, BETA, GAMMA};
    private static enum days {MON, TUE, WED, THU, FRI, SAT, SUN};

    private static greek g;
    private static int m;
    private static days w;
    private static int n;
    private static char grade;
    private static int r;
    private static int a1a[] = new int[10], a1b[] = new int[10];
    private static double a2[] = new double[21];
    private static boolean a3[] = new boolean[4];
    private static greek a4[] = new greek[3];
    private static days a5[][] = new days[21][16];
    private static days a6[][][] = new days[21][16][8];
    private static char chs[] = new char[32];
    private static String str;
    private static String msg[] = new String[21];
    private static int ua1[] = new int[21];
    private static int ua2[][] = new int[21][5];
    private static greek ua3[][][] = new greek[2][21][3];
    private static String us4;
    private static String us5[][] = new String[21][5];

    public static void main(String[] args)
    {
        System.out.println("Done!");
    }
}

File: TestDecls.pas
PROGRAM TestDecls;

CONST
    MINUSFIVE = -5;
    TEN = 10;

TYPE
    greek = (ALPHA, BETA, GAMMA);
    days = (MON, TUE, WED, THU, FRI, SAT, SUN);
    weekdays = MON..FRI;
    sr1 = 5..25;    
    sr2 = MINUSFIVE..TEN;
    sr3 = -TEN..-3;
    
    ar1 = ARRAY [1..TEN] OF integer;
    ar2 = ARRAY [sr1] OF real;
    ar3 = ARRAY [WED..SAT] of boolean;
    ar4 = ARRAY [(SMALL, MEDIUM, LARGE)] OF greek;
    char32 = ARRAY [1..32] OF char;                  
    
    ar5 = ARRAY [sr1] OF ARRAY [sr2] OF days;
    ar6 = ARRAY [sr1, sr2, sr3] OF weekdays;
                        
    string32 = PACKED ARRAY [1..32] OF char;
   
VAR
    g : greek;
    m : sr1;
    w : weekdays;
    
    n : 45..50;
    grade : 'A'..'F';
    r : (IV, V, VI);
    
    a1a, a1b : ar1;
    a2 : ar2;
    a3 : ar3;
    a4 : ar4;
    a5 : ar5;
    a6 : ar6;
    chs : char32;
    str : string32;
    msg : ARRAY [sr1] OF string32;

    ua1 : ARRAY [sr1] OF (JAN, FEB, MAR, APR, MAY);
    ua2 : ARRAY [sr1] OF ARRAY [8..12] OF JAN..APR;
    ua3 : ARRAY [boolean, sr1, (RED, WHITE, BLUE)] 
                                         OF greek;
    us4 : PACKED ARRAY [sr1] OF char;
    us5 : ARRAY [sr1, weekdays] 
                OF PACKED ARRAY [THU..SUN] OF char;
                
BEGIN
END.
